app/api/auth.py

```diff
@@ -294,12 +299,10 @@ def ticket_attendee_authorized(order_identifier):
             key = UPLOAD_PATHS['pdf']['ticket_attendee'].format(identifier=order_identifier)
             file_path = '../generated/tickets/{}/{}/'.format(key, generate_hash(key)) + order_identifier + '.pdf'
             try:
-                response = make_response(send_file(file_path))
-                response.headers['Content-Disposition'] = 'attachment; filename=ticket-%s.zip' % order_identifier
-                return response
+                return return_tickets(file_path, order_identifier)
             except FileNotFoundError:
                 create_pdf_tickets_for_holder(order)
-                return redirect(url_for('ticket_blueprint.ticket_attendee_authorized', order_identifier=order_identifier))
+                return return_tickets(file_path, order_identifier)
```

Comment:
Wouldn't this still result in an infinite loop

Reply:
If the tickets are found, the pdf `response` is returned.
If the tickets are not found, the tickets are generated and the pdf `response` is returned.
Previously, the redirection which was triggered had to hit the whole route again, which resulted in re-execution of the whole function associated with the route.

Reply:
Sorry, I missed this line - `create_pdf_tickets_for_holder(order)`